third_party/terraform/resources/resource_bigquery_table.go.erb

```diff
@@ -859,6 +939,21 @@ func flattenTimePartitioning(tp *bigquery.TimePartitioning) []map[string]interfa
 	return []map[string]interface{}{result}
 }
 
+<% unless version == 'ga' -%>
+func flattenRangePartitioning(rp *bigquery.RangePartitioning) map[string]interface{} {
```

Comment:
this will need to get changed if we change it to list

Reply:
I can change it to a list. What would be preferred?

Reply:
In general, for nested objects like this we actually always want TypeList with MaxItems 1 - TypeMaps actually only work for KV maps (i.e. labels, simple string maps) and it works for nested objects mostly by accident. In the past we used some but are trying to change/remove them as in the future those fields will likely break in newer versions of the Terraform core type system.

For this flattener, it'd just need a []interface{} wrapping the current map[string]interface{}